src/components/shapeStackUtils.js

```diff
@@ -1,15 +1,17 @@
 import { id as poolId } from './nodes/pool';
 import { id as laneId } from './nodes/poolLane';
 
-export default function ensureShapeIsNotCovered(shape) {
+export default function ensureShapeIsNotCovered(shape, graph) {
   if (isPool(shape)) {
     bringPoolToFront(shape);
+    bringFlowsToFront(graph);
   }
 
   const parentPool = getElementPool(shape);
 
   if (parentPool) {
     bringPoolToFront(parentPool);
+    bringFlowsToFront(graph);
```

Comment:
Instead of the duplication here how would you feel above recursively calling this method to achieve this?